### DISCUSSION TITLE
JWT Secret Key Regenerate

### DISCUSSION THREAD

**Discussion by 50l3r** [lexik/LexikJWTAuthenticationBundle#1212] [Q&A] [answered]
Hi folks,

I have a backend which has to be fed by a JWT authentication system that manages another php backend with this library.

Since my backend only manages the verification of the JWT through the secret key, this generates a doubt to me.

In which cases the library generates a new private key?  

I have the only problem that if the private key changes in the php backend, my verification will stop working and I wanted to make sure when this happens.

Thanks in advance :)

**Comment by chalasr** [comment, marked as answer]
Hi,

> In which cases the library generates a new private key?

The bundle itself never regenerates keys, only you can do it. A good reason why you could want to do it is periodic rotation, which the bundle doesn't handle out of the box.

**Reply by 50l3r** [reply]
Much thanks @chalasr
